src/main/java/com/example/schoolmms/repository/product/BrandRepositoryImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.example.schoolmms.repository.product;
+
+import com.example.schoolmms.entity.Brand;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+
+@Repository
+@Transactional(readOnly = true)
+public class BrandRepositoryImpl implements BrandRepository {
+
+    @PersistenceContext
+    EntityManager entityManager;
+
+    @Override
+    public List<Brand> findAll() {
+        CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
```

Comment:
Looks like you have duplicated code in `finaAll` and `findByName`. Try to fix it